.github/workflows/build-src.yml

```diff
@@ -28,14 +28,15 @@ jobs:
         uses: actions/checkout@v4
         with:
           ref: ${{ github.event.pull_request.head.sha }}
-          fetch-depth: 0
+          fetch-depth: 50
```

Comment:
why don't make it `fetch-depth: 1` ?

Reply:
because we have to fetch the entire PR history / up to develop, for stuff like clang-diff-format linter and scripted diffs and whatnot

Reply:
Should we have some mechanism to warn pull requests with >50 commits that they should consolidate or split their PRs to be under this limit?

Reply:
let's put 999 then? 999 is not really a lot (less than 1 major release), but it will cover for sure big PRs
999 should be still fast.

Examples of PR with big amount of commits:
109 commits - https://github.com/dashpay/dash/pull/4752
111 commits - https://github.com/dashpay/dash/pull/4512

Reply:
Works for me, I was going to suggest something lower like 128 since there really is little reason for a PR that large to exist but since we aren't implementing a warning mechanism, 999 is more than enough to cover any PR, sane or otherwise.

Reply:
Honestly; no. I don't ever want to merge a PR more than 50 commits :D